File: backend/Snagged.backend/Snagged.Domain/Entities/Item.cs
using Microsoft.VisualBasic;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Snagged.Domain.Entities
{
    public class Item
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        public int? SubcategoryId { get; set; }
        public Subcategory Subcategory { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public string Condition { get; set; }
        public bool IsSold { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public ICollection<ItemImage> Images { get; set; } = new List<ItemImage>();
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
        public ICollection<Favorite> Favorites { get; set; } = new List<Favorite>();
        public ICollection<Report> Reports { get; set; } = new List<Report>();
        public ICollection<Conversation> Conversations { get; set; } = new List<Conversation>();
    }
}

File: frontend/src/app/shared/models/item.ts
import { User } from './user';
import { Category } from './category';
import { Subcategory } from './subcategory';
import { ItemImage } from './item-image';
import { OrderItem } from './order-item';
import { Favorite } from './favorite';
import { Report } from './report';
import { Conversation } from './conversation';
import { CartItem } from './cart-item';

export interface Item {
  id: number;
  userId: number;
  user?: User;
  categoryId: number;
  category?: Category;
  subcategoryId?: number;
  subcategory?: Subcategory;
  title: string;
  description: string;
  price: number;
  condition: string;
  isSold: boolean;
  createdAt: Date | string;
  images: ItemImage[];
  orderItems: OrderItem[];
  favorites: Favorite[];
  reports: Report[];
  conversations: Conversation[];
  cartItems: CartItem[];
}
